packages/flutter_tools/lib/src/base/error_handling_io.dart

```diff
@@ -542,26 +561,32 @@ void _handlePosixException(Exception e, String message, int errorCode) {
   const int enospc = 28;
   const int eacces = 13;
   // Catch errors and bail when:
+  String errorMessage;
   switch (errorCode) {
     case enospc:
-      throwToolExit(
+      errorMessage =
         '$message. The target device is full.'
         '\n$e\n'
-        'Free up space and try again.',
-      );
+        'Free up space and try again.';
       break;
     case eperm:
     case eacces:
-      throwToolExit(
+      errorMessage =
         '$message. The flutter tool cannot access the file or directory.\n'
         'Please ensure that the SDK and/or project is installed in a location '
-        'that has read/write permissions for the current user.'
-      );
+        'that has read/write permissions for the current user.';
       break;
     default:
       // Caller must rethrow the exception.
       break;
   }
+  if (errorMessage == null) {
```

Comment:
Consider wrapping the logic from here to the `throwToolExit` in a helper method.

Reply:
Done